eknrenderer/eknrenderer/eknr-renderer.c

```diff
@@ -0,0 +1,1311 @@
+                                        GVariant     *variables,
+                                        GError      **error)
+{
+  g_autoptr(GVariantDict) variables_dict = g_variant_dict_new (variables);
+  g_autoptr(RendererMustacheData) data = renderer_mustache_data_new (variables_dict,
+                                                                     error,
+                                                                     tmpl_text);
+  g_autoptr(mustache_template_t) tmpl = mustache_compile (&_renderer_mustache_data_vfuncs,
+                                                          (char *) data);
+
+  if (!tmpl)
+    return NULL;
+
+  return _renderer_render_mustache_document_internal (tmpl, variables, error);
+}
+
+/**
+ * eknr_mustache_template_compiles:
+ * @tmpl_text: The template to render
+ * @error: A #GError
+ *
+ * Attempt to compile a mustache template.
+ *
+ * Returns: The %TRUE on success, %FALSE on error.
+ */
+gboolean
+eknr_mustache_template_compiles (const char   *tmpl_text,
+                                 GError      **error)
+{
+  g_autoptr(RendererMustacheData) data = renderer_mustache_data_new (NULL,
+                                                                     error,
+                                                                     tmpl_text);
+  g_autoptr(mustache_template_t) tmpl = mustache_compile (&_renderer_mustache_data_vfuncs,
+                                                          (char *) data);
+
+  if (!tmpl)
+    return FALSE;
+
+  return TRUE;
+}
+
+static char *
+format_a_href_link (const char   *uri,
+                    const char   *text,
+                    GError      **error)
+{
+  g_autofree char *escaped = eknr_escape_html (text, error);
+
+  if (escaped == NULL)
+    return NULL;
+
+  return g_strdup_printf ("<a class=\"eos-show-link\" href=\"%s\">%s</a>",
+                          uri,
+                          escaped);
+}
+
+static char *
+format_license_link (const char   *license,
+                     GError      **error)
+{
+  g_autofree char *escaped = g_uri_escape_string (license, NULL, TRUE);
+  g_autofree char *licence_link = g_strdup_printf ("license://%s", escaped);
+  return format_a_href_link (licence_link,
+                             eos_get_license_display_name (license),
+                             error);
+}
+
+static GVariant *
+get_legacy_disclaimer_section_content (const char   *source,
+                                       const char   *source_name,
+                                       const char   *original_uri,
+                                       const char   *license,
+                                       const char   *title,
+                                       GError      **error)
+{
+  if (g_strcmp0 (source, "wikisource") == 0 ||
+      g_strcmp0 (source, "wikibooks") == 0 ||
+      g_strcmp0 (source, "wikipedia") == 0)
+    {
+      g_autofree char *original_link = format_a_href_link (original_uri,
```

Comment:
It does not seem like this is used.

Reply:
It is not ... let me check the original code to see if it is needed.

Reply:
Okay, good catch - I was using `source` instead of `original_link`!